flood_adapt/adapter/sfincs_offshore.py

```diff
@@ -0,0 +1,177 @@
+            physical_projection = scenario.projection.get_physical_projection()
+
+            # Create any event specific files
+            event.preprocess(sim_path)
+
+            # Edit offshore model
+            _offshore_model.set_timing(event.attrs.time)
+
+            # Add water levels
+            _offshore_model._add_bzs_from_bca(event, physical_projection.attrs)
+
+            # Add spw if applicable
+            if isinstance(event, HurricaneEvent):
+                _offshore_model._add_forcing_spw(
+                    sim_path / f"{event.attrs.track_name}.spw"
+                )
+
+            # Add wind and if applicable pressure forcing from meteo data
+            elif isinstance(event, HistoricalEvent):
+                wind_forcings = [
+                    f for f in event.get_forcings() if isinstance(f, IWind)
+                ]
+
+                if wind_forcings:
+                    if len(wind_forcings) > 1:
+                        raise ValueError("Only one wind forcing is allowed")
+                    wind_forcing = wind_forcings[0]
+
+                    # Add wind forcing
+                    _offshore_model._add_forcing_wind(wind_forcing)
+
+                    # Add pressure forcing for the offshore model (this doesnt happen normally in _add_forcing_wind() for overland models)
+                    if isinstance(wind_forcing, WindMeteo):
+                        ds = wind_forcing.get_data(
+                            t0=event.attrs.time.start_time, t1=event.attrs.time.end_time
+                        )
+                        _offshore_model._add_pressure_forcing_from_grid(ds=ds)
+
+            # write sfincs model in output destination
+            _offshore_model.write(path_out=sim_path)
+
+    def _execute_sfincs_offshore(self, sim_path: Path, scenario: IScenario):
+        self.logger.info("Running offshore model...")
+
+        with SfincsAdapter(model_root=sim_path) as _offshore_model:
+            if _offshore_model.sfincs_completed(scenario):
+                self.logger.info(
+                    "Skip running offshore model as it has already been run."
+                )
+                return
+
+            success = _offshore_model.execute(path=sim_path, strict=False)
+
+            if not success:
+                raise RuntimeError(
+                    f"Running offshore SFINCS model failed. See {sim_path} for more information."
+                )
+
+    def _get_simulation_path(self, scenario: IScenario) -> Path:
+        event = scenario.strategy
+        if isinstance(event, EventSet):
+            return (
+                db_path(
+                    TopLevelDir.output,
+                    object_dir=ObjectDir.scenario,
+                    obj_name=scenario.attrs.name,
+                )
+                / "Flooding"
+                / "simulations"
+                / event.attrs.name  # ? add sub event name? or do we only run offshore once for the entire event set?
```

Comment:
@GundulaW for event sets using offshore models, since I believe we add the same forcings to all offshore models, we might as well only run one. Not 100% sure though.